packages/react/src/components/MenuButton/MenuButton-test.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { MenuItem } from '../Menu';
+
+import { MenuButton } from './';
+
+describe('MenuButton', () => {
+  it('should support a ref on the outermost element', () => {
+    const ref = jest.fn();
+    const { container } = render(
+      <MenuButton label="Actions" ref={ref}>
+        <MenuItem label="Action" />
+      </MenuButton>
+    );
+    expect(ref).toHaveBeenCalledWith(container.firstChild);
+  });
+
+  it('should support a custom class name on the outermost element', () => {
+    const { container } = render(
+      <MenuButton label="Actions" className="test">
+        <MenuItem label="Action" />
+      </MenuButton>
+    );
+    expect(container.firstChild).toHaveClass('test');
+  });
+
+  it('should forward additional props on the outermost element', () => {
+    const { container } = render(
+      <MenuButton label="Actions" data-testid="test">
+        <MenuItem label="Action" />
+      </MenuButton>
+    );
+    expect(container.firstChild).toHaveAttribute('data-testid', 'test');
+  });
+
+  it('should render props.label on the trigger button', () => {
+    render(
+      <MenuButton label="Test">
+        <MenuItem label="Action" />
+      </MenuButton>
+    );
+    expect(screen.getByRole('button')).toHaveTextContent(/^Test$/);
+  });
+
+  it('should open a menu on click', async () => {
+    render(
+      <MenuButton label="Actions">
+        <MenuItem label="Action" />
+      </MenuButton>
+    );
+
+    await userEvent.click(screen.getByRole('button'));
+
+    expect(screen.getByRole('menu')).toBeTruthy();
+    expect(screen.getByRole('menuitem')).toHaveTextContent(/^Action$/);
+  });
+
+  it('should support being disabled', () => {
+    render(
+      <MenuButton label="Actions" disabled>
+        <MenuItem label="Action" />
+      </MenuButton>
+    );
+
+    expect(screen.getByRole('button')).toBeDisabled();
+  });
+});
```

Comment:
should this file also verify for the [`children`](https://github.com/carbon-design-system/carbon/pull/13204/files#diff-051197835517702352195445ea71c7fdf1322b984526104996a020a6eb16a87dR42-R56), [`size`](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/DataTable/__tests__/Table-test.js#LL12-L19C5), and `kind` props? 

A [snapshot test](https://github.com/carbon-design-system/carbon/pull/13204/files#diff-051197835517702352195445ea71c7fdf1322b984526104996a020a6eb16a87dR15-R26) at the top might also be nice

Can we also separate the tests into "'renders as expected - Component API'" and "'behaves as expected'" categories to follow standard? See example file: https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/DataTable/__tests__/TableHeader-test.js

Reply:
Added tests for `size` and `kind`. Rest same comment as above.